code/processes/dqe.q

```diff
@@ -6,48 +6,81 @@ detailcsv:@[value;`.dqe.detailcsv;first .proc.getconfigfile["dqedetail.csv"]];
+    .dqe.updresultstab[runtype;idnum;0Np;0b;"error:fail to complete before next run";`failed;params;proc]];
   }
 
-initstatusupd:{[runtype;idnum;funct;vars;rs]                                                                   /- set initial values in results table
+initstatusupd:{[runtype;idnum;funct;params;rs]                                                                  /- set initial values in results table
+  if[idnum in exec id from .dqe.compcounter;delete from `.dqe.compcounter where id=idnum;];
   .lg.o[`initstatus;"setting up initial record(s) for id ",(string idnum)];
-  .dqe.dupchk[runtype;idnum]'[rs];                                                                             /- calls dupchk function to check if last runs chkstatus is still started
-  `.dqe.results insert (idnum;funct;`$"," sv string raze (),vars;rs[0];rs[1];.z.p;0Np;0b;"";`started;runtype);
+  .dqe.dupchk[runtype;idnum;params]'[rs];                                                                       /- calls dupchk function to check if last runs chkstatus is still started
+  `.dqe.results insert (idnum;funct;`$"," sv string raze (),4#params;rs[0];rs[1];.z.p;0Np;0b;"";`started;runtype);
+  }
+
+updresultstab:{[runtype;idnum;end;res;des;status;params;proc]                                                   /- general function used to update a check in the results table
+  .lg.o[`updresultstab;"Updating check id ",(string idnum)," in the results table with status ",string status];
+  if[1b=params`comp;proc:params`compresproc];
+  if[c:count select from .dqe.results where id=idnum, procschk=proc,chkstatus=`started;                         /- obtain count of checks that will be updated
+    .lg.o[`updresultstab;raze "run check id ",(string idnum)," update in results table with check status ",string status];
+    `.dqe.results set update endtime:end,result:res,descp:enlist des,chkstatus:status,chkruntype:runtype from .dqe.results where id=idnum,procschk=proc,chkstatus=`started];
+  delete from `.dqe.compcounter where id=idnum;
+  params:()!();
   }
 
-failchk:{[runtype;idnum;error;proc]                                                                            /- general fail function, used to fail a check with inputted error message
-  c:count select from .dqe.results where id=idnum, procschk=proc,chkstatus=`started;
-  if[c;.lg.o[`failerr;raze "run check id ",(string idnum)," update in results table with a fail, with ",(string error)]];
-  `.dqe.results set update chkstatus:`failed,output:0b,descp:c#enlist error,chkruntype:runtype from .dqe.results where id=idnum, procschk=proc,chkstatus=`started;
+chkcompare:{[runtype;idnum;params]                                                                              /- function to compare the checks
+  if[not params[`compcount]=.dqe.compcounter[idnum][`counter];:()];                                             /- checks if all async check results have returned
```

Comment:
One minor thing for this section, I don't think `.dqe.compcounter` is being updated but is used over and over again so could be saved as a variable on this line and reused just to cut some bulk:
```suggestion
  if[params[`compcount]<>(d:.dqe.compcounter idnum)`counter;:()];                                               /- checks if all async check results have returned
```